identity-wallet/src/state/reducers/dev_mode.rs

```diff
@@ -38,20 +42,248 @@ lazy_static! {
     );
 }
 
-pub async fn set_dev_mode(state: AppState, action: Action) -> Result<AppState, AppError> {
-    if let Some(enabled) = listen::<SetDevMode>(action).map(|payload| payload.enabled) {
-        return Ok(AppState {
-            dev_mode_enabled: enabled,
-            current_user_prompt: None,
-            ..state
-        });
+const PASSWORD: &str = "sup3rSecr3t";
+
+pub async fn load_dev_profile(state: AppState, action: Action) -> Result<AppState, AppError> {
+    info!("Load dev profile: {:?}", action);
+
+    if let Some(dev_profile) = listen::<DevProfile>(action) {
+        match dev_profile.profile {
+            ProfileType::Ferris => return load_ferris_profile().await,
+            ProfileType::Dragon => return load_dragon_profile(state).await,
+            ProfileType::None => {}
+        }
     }
+
     Ok(state)
 }
 
-pub async fn load_dev_profile(_state: AppState, _action: Action) -> Result<AppState, AppError> {
-    info!("load dev profile");
+pub async fn toggle_dev_settings(mut state: AppState, _action: Action) -> Result<AppState, AppError> {
+    info!("Toggle dev settings");
+
+    if state.dev_profile.is_some() {
+        state.dev_profile = None;
+    } else {
+        state.dev_profile = Some(ProfileType::None);
+    }
+
+    state.current_user_prompt = None;
+
+    Ok(state)
+}
+
+pub async fn login_profile(state: AppState) -> Result<AppState, AppError> {
+    command::reduce(
+        state,
+        Arc::new(UnlockStorage {
+            password: PASSWORD.to_string(),
```

Comment:
Since that `login_profile()` function is called for whenever `dev_profile != Some(ProfileType::None)`, that implies that _all_ dev profiles share the same password (which is okay since it's public anyway). We need to make sure that all dev profiles use the `const PASSWORD`.

Reply:
Yes I will make a comment in the load_dev_profile function.

Reply:
If it's now a requirement, could you also apply that change and put the `PASSWORD` in `load_ferris_profile()` in L145?